// ngx-dropzone-wrapper/bundles/ngx-dropzone-wrapper.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('dropzone')) :
    typeof define === 'function' && define.amd ? define('ngx-dropzone-wrapper', ['exports', '@angular/core', '@angular/common', 'dropzone'], factory) :
    (global = global || self, factory(global['ngx-dropzone-wrapper'] = {}, global.ng.core, global.ng.common, global.Dropzone));
}(this, (function (exports, core, common, Dropzone) { 'use strict';

    Dropzone = Dropzone && Dropzone.hasOwnProperty('default') ? Dropzone['default'] : Dropzone;

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    var DROPZONE_CONFIG = new core.InjectionToken('DROPZONE_CONFIG');
    var DropzoneEvents = [
        'error',
        'success',
        'sending',
        'canceled',
        'complete',
        'processing',
        'drop',
        'dragStart',
        'dragEnd',
        'dragEnter',
        'dragOver',
        'dragLeave',
        'thumbnail',
        'addedFile',
        'removedFile',
        'uploadProgress',
        'maxFilesReached',
        'maxFilesExceeded',
        'successMultiple',
        'sendingMultiple',
        'canceledMultiple',
        'completeMultiple',
        'processingMultiple',
        'reset',
        'queueComplete',
        'totalUploadProgress'
    ];
    var DropzoneConfig = /** @class */ (function () {
        function DropzoneConfig(config) {
            if (config === void 0) { config = {}; }
            this.assign(config);
        }
        DropzoneConfig.prototype.assign = function (config, target) {
            if (config === void 0) { config = {}; }
            target = target || this;
            for (var key in config) {
                if (config[key] != null && !(Array.isArray(config[key])) &&
                    typeof config[key] === 'object' && !(config[key] instanceof HTMLElement)) {
                    target[key] = {};
                    this.assign(config[key], target[key]);
                }
                else {
                    target[key] = config[key];
                }
            }
        };
        return DropzoneConfig;
    }());

    var DropzoneDirective = /** @class */ (function () {
        function DropzoneDirective(zone, renderer, elementRef, differs, platformId, defaults) {
            this.zone = zone;
            this.renderer = renderer;
            this.elementRef = elementRef;
            this.differs = differs;
            this.platformId = platformId;
            this.defaults = defaults;
            this.configDiff = null;
            this.disabled = false;
            this.DZ_INIT = new core.EventEmitter();
            this.DZ_ERROR = new core.EventEmitter();
            this.DZ_SUCCESS = new core.EventEmitter();
            this.DZ_SENDING = new core.EventEmitter();
            this.DZ_CANCELED = new core.EventEmitter();
            this.DZ_COMPLETE = new core.EventEmitter();
            this.DZ_PROCESSING = new core.EventEmitter();
            this.DZ_DROP = new core.EventEmitter();
            this.DZ_DRAGSTART = new core.EventEmitter();
            this.DZ_DRAGEND = new core.EventEmitter();
            this.DZ_DRAGENTER = new core.EventEmitter();
            this.DZ_DRAGOVER = new core.EventEmitter();
            this.DZ_DRAGLEAVE = new core.EventEmitter();
            this.DZ_THUMBNAIL = new core.EventEmitter();
            this.DZ_ADDEDFILE = new core.EventEmitter();
            this.DZ_REMOVEDFILE = new core.EventEmitter();
            this.DZ_UPLOADPROGRESS = new core.EventEmitter();
            this.DZ_MAXFILESREACHED = new core.EventEmitter();
            this.DZ_MAXFILESEXCEEDED = new core.EventEmitter();
            this.DZ_SUCCESSMULTIPLE = new core.EventEmitter();
            this.DZ_SENDINGMULTIPLE = new core.EventEmitter();
            this.DZ_CANCELEDMULTIPLE = new core.EventEmitter();
            this.DZ_COMPLETEMULTIPLE = new core.EventEmitter();
            this.DZ_PROCESSINGMULTIPLE = new core.EventEmitter();
            this.DZ_RESET = new core.EventEmitter();
            this.DZ_QUEUECOMPLETE = new core.EventEmitter();
            this.DZ_TOTALUPLOADPROGRESS = new core.EventEmitter();
            var dz = Dropzone;
            dz.autoDiscover = false;
        }
        DropzoneDirective.prototype.ngOnInit = function () {
            var _this = this;
            if (!common.isPlatformBrowser(this.platformId)) {
                return;
            }
            var params = new DropzoneConfig(this.defaults);
            params.assign(this.config); // Custom configuration
            this.renderer.addClass(this.elementRef.nativeElement, (params.maxFiles === 1) ? 'dz-single' : 'dz-multiple');
            this.renderer.removeClass(this.elementRef.nativeElement, (params.maxFiles === 1) ? 'dz-multiple' : 'dz-single');
            this.zone.runOutsideAngular(function () {
                _this.instance = new Dropzone(_this.elementRef.nativeElement, params);
            });
            if (this.disabled) {
                this.instance.disable();
            }
            if (this.DZ_INIT.observers.length) {
                this.zone.run(function () {
                    _this.DZ_INIT.emit(_this.instance);
                });
            }
            // Add auto reset handling for events
            this.instance.on('success', function () {
                if (params.autoReset != null) {
                    setTimeout(function () { return _this.reset(); }, params.autoReset);
                }
            });
            this.instance.on('error', function () {
                if (params.errorReset != null) {
                    setTimeout(function () { return _this.reset(); }, params.errorReset);
                }
            });
            this.instance.on('canceled', function () {
                if (params.cancelReset != null) {
                    setTimeout(function () { return _this.reset(); }, params.cancelReset);
                }
            });
            // Add native Dropzone event handling
            DropzoneEvents.forEach(function (eventName) {
                _this.instance.on(eventName.toLowerCase(), function () {
                    var args = [];
                    for (var _i = 0; _i < arguments.length; _i++) {
                        args[_i] = arguments[_i];
                    }
                    args = (args.length === 1) ? args[0] : args;
                    var output = "DZ_" + eventName.toUpperCase();
                    var emitter = _this[output];
                    if (emitter.observers.length > 0) {
                        _this.zone.run(function () {
                            emitter.emit(args);
                        });
                    }
                });
            });
            if (!this.configDiff) {
                this.configDiff = this.differs.find(this.config || {}).create();
                this.configDiff.diff(this.config || {});
            }
        };
        DropzoneDirective.prototype.ngOnDestroy = function () {
            var _this = this;
            if (this.instance) {
                this.zone.runOutsideAngular(function () {
                    _this.instance.destroy();
                });
                this.instance = null;
            }
        };
        DropzoneDirective.prototype.ngDoCheck = function () {
            if (!this.disabled && this.configDiff) {
                var changes = this.configDiff.diff(this.config || {});
                if (changes && this.instance) {
                    this.ngOnDestroy();
                    this.ngOnInit();
                }
            }
        };
        DropzoneDirective.prototype.ngOnChanges = function (changes) {
            var _this = this;
            if (this.instance && changes['disabled']) {
                if (changes['disabled'].currentValue !== changes['disabled'].previousValue) {
                    if (changes['disabled'].currentValue === false) {
                        this.zone.runOutsideAngular(function () {
                            _this.instance.enable();
                        });
                    }
                    else if (changes['disabled'].currentValue === true) {
                        this.zone.runOutsideAngular(function () {
                            _this.instance.disable();
                        });
                    }
                }
            }
        };
        DropzoneDirective.prototype.dropzone = function () {
            return this.instance;
        };
        DropzoneDirective.prototype.reset = function (cancel) {
            var _this = this;
            if (this.instance) {
                this.zone.runOutsideAngular(function () {
                    _this.instance.removeAllFiles(cancel);
                });
            }
        };
        DropzoneDirective.ctorParameters = function () { return [
            { type: core.NgZone },
            { type: core.Renderer2 },
            { type: core.ElementRef },
            { type: core.KeyValueDiffers },
            { type: Object, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] },
            { type: undefined, decorators: [{ type: core.Optional }, { type: core.Inject, args: [DROPZONE_CONFIG,] }] }
        ]; };
        __decorate([
            core.Input()
        ], DropzoneDirective.prototype, "disabled", void 0);
        __decorate([
            core.Input('dropzone')
        ], DropzoneDirective.prototype, "config", void 0);
        __decorate([
            core.Output('init')
        ], DropzoneDirective.prototype, "DZ_INIT", void 0);
        __decorate([
            core.Output('error')
        ], DropzoneDirective.prototype, "DZ_ERROR", void 0);
        __decorate([
            core.Output('success')
        ], DropzoneDirective.prototype, "DZ_SUCCESS", void 0);
        __decorate([
            core.Output('sending')
        ], DropzoneDirective.prototype, "DZ_SENDING", void 0);
        __decorate([
            core.Output('canceled')
        ], DropzoneDirective.prototype, "DZ_CANCELED", void 0);
        __decorate([
            core.Output('complete')
        ], DropzoneDirective.prototype, "DZ_COMPLETE", void 0);
        __decorate([
            core.Output('processing')
        ], DropzoneDirective.prototype, "DZ_PROCESSING", void 0);
        __decorate([
            core.Output('drop')
        ], DropzoneDirective.prototype, "DZ_DROP", void 0);
        __decorate([
            core.Output('dragStart')
        ], DropzoneDirective.prototype, "DZ_DRAGSTART", void 0);
        __decorate([
            core.Output('dragEnd')
        ], DropzoneDirective.prototype, "DZ_DRAGEND", void 0);
        __decorate([
            core.Output('dragEnter')
        ], DropzoneDirective.prototype, "DZ_DRAGENTER", void 0);
        __decorate([
            core.Output('dragOver')
        ], DropzoneDirective.prototype, "DZ_DRAGOVER", void 0);
        __decorate([
            core.Output('dragLeave')
        ], DropzoneDirective.prototype, "DZ_DRAGLEAVE", void 0);
        __decorate([
            core.Output('thumbnail')
        ], DropzoneDirective.prototype, "DZ_THUMBNAIL", void 0);
        __decorate([
            core.Output('addedFile')
        ], DropzoneDirective.prototype, "DZ_ADDEDFILE", void 0);
        __decorate([
            core.Output('removedFile')
        ], DropzoneDirective.prototype, "DZ_REMOVEDFILE", void 0);
        __decorate([
            core.Output('uploadProgress')
        ], DropzoneDirective.prototype, "DZ_UPLOADPROGRESS", void 0);
        __decorate([
            core.Output('maxFilesReached')
        ], DropzoneDirective.prototype, "DZ_MAXFILESREACHED", void 0);
        __decorate([
            core.Output('maxFilesExceeded')
        ], DropzoneDirective.prototype, "DZ_MAXFILESEXCEEDED", void 0);
        __decorate([
            core.Output('successMultiple')
        ], DropzoneDirective.prototype, "DZ_SUCCESSMULTIPLE", void 0);
        __decorate([
            core.Output('sendingMultiple')
        ], DropzoneDirective.prototype, "DZ_SENDINGMULTIPLE", void 0);
        __decorate([
            core.Output('canceledMultiple')
        ], DropzoneDirective.prototype, "DZ_CANCELEDMULTIPLE", void 0);
        __decorate([
            core.Output('completeMultiple')
        ], DropzoneDirective.prototype, "DZ_COMPLETEMULTIPLE", void 0);
        __decorate([
            core.Output('processingMultiple')
        ], DropzoneDirective.prototype, "DZ_PROCESSINGMULTIPLE", void 0);
        __decorate([
            core.Output('reset')
        ], DropzoneDirective.prototype, "DZ_RESET", void 0);
        __decorate([
            core.Output('queueComplete')
        ], DropzoneDirective.prototype, "DZ_QUEUECOMPLETE", void 0);
        __decorate([
            core.Output('totalUploadProgress')
        ], DropzoneDirective.prototype, "DZ_TOTALUPLOADPROGRESS", void 0);
        DropzoneDirective = __decorate([
            core.Directive({
                selector: '[dropzone]',
                exportAs: 'ngxDropzone'
            }),
            __param(4, core.Inject(core.PLATFORM_ID)),
            __param(5, core.Optional()), __param(5, core.Inject(DROPZONE_CONFIG))
        ], DropzoneDirective);
        return DropzoneDirective;
    }());

    var DropzoneComponent = /** @class */ (function () {
        function DropzoneComponent(platformId) {
            this.platformId = platformId;
            this.disabled = false;
            this.message = 'Click or drag files to upload';
            this.placeholder = '';
            this.useDropzoneClass = true;
            this.DZ_INIT = new core.EventEmitter();
            this.DZ_ERROR = new core.EventEmitter();
            this.DZ_SUCCESS = new core.EventEmitter();
            this.DZ_SENDING = new core.EventEmitter();
            this.DZ_CANCELED = new core.EventEmitter();
            this.DZ_COMPLETE = new core.EventEmitter();
            this.DZ_PROCESSING = new core.EventEmitter();
            this.DZ_DROP = new core.EventEmitter();
            this.DZ_DRAGSTART = new core.EventEmitter();
            this.DZ_DRAGEND = new core.EventEmitter();
            this.DZ_DRAGENTER = new core.EventEmitter();
            this.DZ_DRAGOVER = new core.EventEmitter();
            this.DZ_DRAGLEAVE = new core.EventEmitter();
            this.DZ_THUMBNAIL = new core.EventEmitter();
            this.DZ_ADDEDFILE = new core.EventEmitter();
            this.DZ_REMOVEDFILE = new core.EventEmitter();
            this.DZ_UPLOADPROGRESS = new core.EventEmitter();
            this.DZ_MAXFILESREACHED = new core.EventEmitter();
            this.DZ_MAXFILESEXCEEDED = new core.EventEmitter();
            this.DZ_SUCCESSMULTIPLE = new core.EventEmitter();
            this.DZ_SENDINGMULTIPLE = new core.EventEmitter();
            this.DZ_CANCELEDMULTIPLE = new core.EventEmitter();
            this.DZ_COMPLETEMULTIPLE = new core.EventEmitter();
            this.DZ_PROCESSINGMULTIPLE = new core.EventEmitter();
            this.DZ_RESET = new core.EventEmitter();
            this.DZ_QUEUECOMPLETE = new core.EventEmitter();
            this.DZ_TOTALUPLOADPROGRESS = new core.EventEmitter();
        }
        DropzoneComponent.prototype.ngOnInit = function () {
            var _this = this;
            if (!common.isPlatformBrowser(this.platformId)) {
                return;
            }
            window.setTimeout(function () {
                DropzoneEvents.forEach(function (eventName) {
                    if (_this.directiveRef) {
                        var output = "DZ_" + eventName.toUpperCase();
                        var directiveOutput = output;
                        var componentOutput = output;
                        _this.directiveRef[directiveOutput] = _this[componentOutput];
                    }
                });
            }, 0);
        };
        DropzoneComponent.prototype.getPlaceholder = function () {
            return 'url(' + encodeURI(this.placeholder) + ')';
        };
        DropzoneComponent.ctorParameters = function () { return [
            { type: Object, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] }
        ]; };
        __decorate([
            core.Input()
        ], DropzoneComponent.prototype, "disabled", void 0);
        __decorate([
            core.Input()
        ], DropzoneComponent.prototype, "config", void 0);
        __decorate([
            core.Input()
        ], DropzoneComponent.prototype, "message", void 0);
        __decorate([
            core.Input()
        ], DropzoneComponent.prototype, "placeholder", void 0);
        __decorate([
            core.Input()
        ], DropzoneComponent.prototype, "useDropzoneClass", void 0);
        __decorate([
            core.Output('init')
        ], DropzoneComponent.prototype, "DZ_INIT", void 0);
        __decorate([
            core.Output('error')
        ], DropzoneComponent.prototype, "DZ_ERROR", void 0);
        __decorate([
            core.Output('success')
        ], DropzoneComponent.prototype, "DZ_SUCCESS", void 0);
        __decorate([
            core.Output('sending')
        ], DropzoneComponent.prototype, "DZ_SENDING", void 0);
        __decorate([
            core.Output('canceled')
        ], DropzoneComponent.prototype, "DZ_CANCELED", void 0);
        __decorate([
            core.Output('complete')
        ], DropzoneComponent.prototype, "DZ_COMPLETE", void 0);
        __decorate([
            core.Output('processing')
        ], DropzoneComponent.prototype, "DZ_PROCESSING", void 0);
        __decorate([
            core.Output('drop')
        ], DropzoneComponent.prototype, "DZ_DROP", void 0);
        __decorate([
            core.Output('dragStart')
        ], DropzoneComponent.prototype, "DZ_DRAGSTART", void 0);
        __decorate([
            core.Output('dragEnd')
        ], DropzoneComponent.prototype, "DZ_DRAGEND", void 0);
        __decorate([
            core.Output('dragEnter')
        ], DropzoneComponent.prototype, "DZ_DRAGENTER", void 0);
        __decorate([
            core.Output('dragOver')
        ], DropzoneComponent.prototype, "DZ_DRAGOVER", void 0);
        __decorate([
            core.Output('dragLeave')
        ], DropzoneComponent.prototype, "DZ_DRAGLEAVE", void 0);
        __decorate([
            core.Output('thumbnail')
        ], DropzoneComponent.prototype, "DZ_THUMBNAIL", void 0);
        __decorate([
            core.Output('addedFile')
        ], DropzoneComponent.prototype, "DZ_ADDEDFILE", void 0);
        __decorate([
            core.Output('removedFile')
        ], DropzoneComponent.prototype, "DZ_REMOVEDFILE", void 0);
        __decorate([
            core.Output('uploadProgress')
        ], DropzoneComponent.prototype, "DZ_UPLOADPROGRESS", void 0);
        __decorate([
            core.Output('maxFilesReached')
        ], DropzoneComponent.prototype, "DZ_MAXFILESREACHED", void 0);
        __decorate([
            core.Output('maxFilesExceeded')
        ], DropzoneComponent.prototype, "DZ_MAXFILESEXCEEDED", void 0);
        __decorate([
            core.Output('successMultiple')
        ], DropzoneComponent.prototype, "DZ_SUCCESSMULTIPLE", void 0);
        __decorate([
            core.Output('sendingMultiple')
        ], DropzoneComponent.prototype, "DZ_SENDINGMULTIPLE", void 0);
        __decorate([
            core.Output('canceledMultiple')
        ], DropzoneComponent.prototype, "DZ_CANCELEDMULTIPLE", void 0);
        __decorate([
            core.Output('completeMultiple')
        ], DropzoneComponent.prototype, "DZ_COMPLETEMULTIPLE", void 0);
        __decorate([
            core.Output('processingMultiple')
        ], DropzoneComponent.prototype, "DZ_PROCESSINGMULTIPLE", void 0);
        __decorate([
            core.Output('reset')
        ], DropzoneComponent.prototype, "DZ_RESET", void 0);
        __decorate([
            core.Output('queueComplete')
        ], DropzoneComponent.prototype, "DZ_QUEUECOMPLETE", void 0);
        __decorate([
            core.Output('totalUploadProgress')
        ], DropzoneComponent.prototype, "DZ_TOTALUPLOADPROGRESS", void 0);
        __decorate([
            core.ViewChild(DropzoneDirective, { static: true })
        ], DropzoneComponent.prototype, "directiveRef", void 0);
        DropzoneComponent = __decorate([
            core.Component({
                selector: 'dropzone',
                exportAs: 'ngxDropzone',
                template: "<div class=\"dz-wrapper\" [class.dropzone]=\"useDropzoneClass\" [dropzone]=\"config\" [disabled]=\"disabled\" (init)=\"DZ_INIT.emit($event)\">\n  <div class=\"dz-message\" [class.disabled]=\"disabled\" [class.dz-placeholder]=\"placeholder\">\n    <div class=\"dz-text\" [innerHTML]=\"config?.dictDefaultMessage || message\"></div>\n\n    <div *ngIf=\"placeholder\" class=\"dz-image\" [style.background-image]=\"getPlaceholder()\"></div>\n  </div>\n\n  <ng-content></ng-content>\n</div>\n",
                encapsulation: core.ViewEncapsulation.None,
                styles: ["dropzone{display:block;width:100%;height:auto}dropzone[fxflex]{display:-webkit-box;display:flex;flex-direction:inherit;min-width:0;min-height:0;-webkit-box-direction:inherit;-webkit-box-orient:inherit}dropzone[fxflex]>.dropzone.dz-wrapper{flex:1 1 auto;min-width:0;min-height:0;-webkit-box-flex:1}dropzone[fxlayout]{align-items:inherit;align-content:inherit;justify-content:inherit;-webkit-box-align:inherit;-webkit-box-pack:inherit}dropzone[fxlayout]>.dropzone.dz-wrapper.dz-single{display:-webkit-box;display:flex;flex-direction:column;align-items:center;align-content:center;justify-content:center;-webkit-box-align:center;-webkit-box-direction:column;-webkit-box-orient:column;-webkit-box-pack:center}dropzone[fxlayout]>.dropzone.dz-wrapper.dz-multiple{display:-webkit-box;display:flex;flex-flow:row wrap;-webkit-box-align:start;align-items:flex-start;align-content:flex-start;-webkit-box-pack:justify;justify-content:space-between;-webkit-box-align:flex-start;-webkit-box-direction:row;-webkit-box-orient:row;-webkit-box-pack:flex-start}dropzone>.dropzone.dz-wrapper{position:relative;overflow:auto;width:100%;min-height:0;max-height:100%;padding:0;border:none;color:#666;background:0 0}dropzone>.dropzone.dz-wrapper .dz-message{position:relative;display:inline-block;overflow:auto;width:calc(100% - 16px);min-width:calc(100% - 16px);min-height:40px;max-width:calc(100% - 16px);max-height:100%;margin:8px;border:2px dashed #aaa;background-color:#eee}dropzone>.dropzone.dz-wrapper .dz-message .dz-text{position:absolute;top:50%;width:100%;padding:8px 16px;text-align:center;-webkit-transform:translateY(-50%);transform:translateY(-50%)}dropzone>.dropzone.dz-wrapper .dz-message .dz-image{width:100%;height:100%;background-size:contain;background-repeat:no-repeat;background-position:50% 50%}dropzone>.dropzone.dz-wrapper .dz-message.disabled{cursor:not-allowed}dropzone>.dropzone.dz-wrapper .dz-message.disabled .dz-text{opacity:.5}dropzone>.dropzone.dz-wrapper .dz-message.dz-placeholder{border-color:rgba(#aaa,0)}dropzone>.dropzone.dz-wrapper .dz-message.dz-placeholder .dz-text{position:absolute;z-index:1;top:0;right:10%;left:10%;opacity:0;font-weight:700;background-color:rgba(#fff,.5);-webkit-transform:translateY(-50%);transform:translateY(-50%);-webkit-transition:opacity 250ms ease-in-out,border-color 250ms ease-in-out,-webkit-filter 250ms ease-in-out;transition:filter 250ms ease-in-out,opacity 250ms ease-in-out,border-color 250ms ease-in-out,-webkit-filter 250ms ease-in-out}dropzone>.dropzone.dz-wrapper .dz-message.dz-placeholder:hover:not(.disabled){border-color:#aaa}dropzone>.dropzone.dz-wrapper .dz-message.dz-placeholder:hover:not(.disabled) .dz-text{opacity:1}dropzone>.dropzone.dz-wrapper .dz-message.dz-placeholder:hover:not(.disabled) .dz-image{-webkit-filter:blur(8px);filter:blur(8px)}dropzone>.dropzone.dz-wrapper .dz-preview{margin:8px}dropzone>.dropzone.dz-wrapper .dz-preview .dz-details{padding:24px}dropzone>.dropzone.dz-wrapper .dz-preview .dz-progress{width:80%;margin-left:-40%;border:1px solid #aaa;border-radius:4px}dropzone>.dropzone.dz-wrapper .dz-preview .dz-progress .dz-upload{background-color:#666}dropzone>.dropzone.dz-wrapper .dz-preview .dz-filename span{display:block;overflow:hidden;width:100%;max-width:100%;text-overflow:ellipsis}dropzone>.dropzone.dz-wrapper .dz-preview .dz-filename span:hover{overflow:visible;white-space:normal;word-wrap:break-word}dropzone>.dropzone.dz-wrapper.dz-single .dz-message{width:100%;height:100%}dropzone>.dropzone.dz-wrapper.dz-single.dz-started .dz-message{display:none}dropzone>.dropzone.dz-wrapper.dz-single .dz-preview{width:calc(100% - 16px);height:100%}dropzone>.dropzone.dz-wrapper.dz-single .dz-preview .dz-image{width:100%;height:100%;border-radius:0}dropzone>.dropzone.dz-wrapper.dz-single .dz-preview .dz-image img{display:block;width:100%;height:auto;margin:0}dropzone>.dropzone.dz-wrapper.dz-single .dz-error-message{top:50%;left:50%;-webkit-transform:translateX(-50%) translateY(100%);transform:translateX(-50%) translateY(100%)}dropzone>.dropzone.dz-wrapper.dz-multiple.dz-started .dz-message{display:inline-block}", "@-webkit-keyframes passing-through{0%{opacity:0;-webkit-transform:translateY(40px);transform:translateY(40px)}30%,70%{opacity:1;-webkit-transform:translateY(0);transform:translateY(0)}100%{opacity:0;-webkit-transform:translateY(-40px);transform:translateY(-40px)}}@keyframes passing-through{0%{opacity:0;-webkit-transform:translateY(40px);transform:translateY(40px)}30%,70%{opacity:1;-webkit-transform:translateY(0);transform:translateY(0)}100%{opacity:0;-webkit-transform:translateY(-40px);transform:translateY(-40px)}}@-webkit-keyframes slide-in{0%{opacity:0;-webkit-transform:translateY(40px);transform:translateY(40px)}30%{opacity:1;-webkit-transform:translateY(0);transform:translateY(0)}}@keyframes slide-in{0%{opacity:0;-webkit-transform:translateY(40px);transform:translateY(40px)}30%{opacity:1;-webkit-transform:translateY(0);transform:translateY(0)}}@-webkit-keyframes pulse{0%,20%{-webkit-transform:scale(1);transform:scale(1)}10%{-webkit-transform:scale(1.1);transform:scale(1.1)}}@keyframes pulse{0%,20%{-webkit-transform:scale(1);transform:scale(1)}10%{-webkit-transform:scale(1.1);transform:scale(1.1)}}.dropzone,.dropzone *{box-sizing:border-box}.dropzone{min-height:150px;border:2px solid rgba(0,0,0,.3);background:#fff;padding:20px}.dropzone.dz-clickable{cursor:pointer}.dropzone.dz-clickable *{cursor:default}.dropzone.dz-clickable .dz-message,.dropzone.dz-clickable .dz-message *{cursor:pointer}.dropzone.dz-started .dz-message{display:none}.dropzone.dz-drag-hover{border-style:solid}.dropzone.dz-drag-hover .dz-message{opacity:.5}.dropzone .dz-message{text-align:center;margin:2em 0}.dropzone .dz-preview{position:relative;display:inline-block;vertical-align:top;margin:16px;min-height:100px}.dropzone .dz-preview:hover{z-index:1000}.dropzone .dz-preview.dz-file-preview .dz-image{border-radius:20px;background:#999;background:-webkit-gradient(linear,left top,left bottom,from(#eee),to(#ddd));background:linear-gradient(to bottom,#eee,#ddd)}.dropzone .dz-preview.dz-file-preview .dz-details{opacity:1}.dropzone .dz-preview.dz-image-preview{background:#fff}.dropzone .dz-preview.dz-image-preview .dz-details{-webkit-transition:opacity .2s linear;transition:opacity .2s linear}.dropzone .dz-preview .dz-remove{font-size:14px;text-align:center;display:block;cursor:pointer;border:none}.dropzone .dz-preview .dz-remove:hover{text-decoration:underline}.dropzone .dz-preview:hover .dz-details{opacity:1}.dropzone .dz-preview .dz-details{z-index:20;position:absolute;top:0;left:0;opacity:0;font-size:13px;min-width:100%;max-width:100%;padding:2em 1em;text-align:center;color:rgba(0,0,0,.9);line-height:150%}.dropzone .dz-preview .dz-details .dz-size{margin-bottom:1em;font-size:16px}.dropzone .dz-preview .dz-details .dz-filename{white-space:nowrap}.dropzone .dz-preview .dz-details .dz-filename:hover span{border:1px solid rgba(200,200,200,.8);background-color:rgba(255,255,255,.8)}.dropzone .dz-preview .dz-details .dz-filename:not(:hover){overflow:hidden;text-overflow:ellipsis}.dropzone .dz-preview .dz-details .dz-filename:not(:hover) span{border:1px solid transparent}.dropzone .dz-preview .dz-details .dz-filename span,.dropzone .dz-preview .dz-details .dz-size span{background-color:rgba(255,255,255,.4);padding:0 .4em;border-radius:3px}.dropzone .dz-preview:hover .dz-image img{-webkit-transform:scale(1.05,1.05);transform:scale(1.05,1.05);-webkit-filter:blur(8px);filter:blur(8px)}.dropzone .dz-preview .dz-image{border-radius:20px;overflow:hidden;width:120px;height:120px;position:relative;display:block;z-index:10}.dropzone .dz-preview .dz-image img{display:block}.dropzone .dz-preview.dz-success .dz-success-mark{-webkit-animation:3s cubic-bezier(.77,0,.175,1) passing-through;animation:3s cubic-bezier(.77,0,.175,1) passing-through}.dropzone .dz-preview.dz-error .dz-error-mark{opacity:1;-webkit-animation:3s cubic-bezier(.77,0,.175,1) slide-in;animation:3s cubic-bezier(.77,0,.175,1) slide-in}.dropzone .dz-preview .dz-error-mark,.dropzone .dz-preview .dz-success-mark{pointer-events:none;opacity:0;z-index:500;position:absolute;display:block;top:50%;left:50%;margin-left:-27px;margin-top:-27px}.dropzone .dz-preview .dz-error-mark svg,.dropzone .dz-preview .dz-success-mark svg{display:block;width:54px;height:54px}.dropzone .dz-preview.dz-processing .dz-progress{opacity:1;-webkit-transition:.2s linear;transition:.2s linear}.dropzone .dz-preview.dz-complete .dz-progress{opacity:0;-webkit-transition:opacity .4s ease-in;transition:opacity .4s ease-in}.dropzone .dz-preview:not(.dz-processing) .dz-progress{-webkit-animation:6s infinite pulse;animation:6s infinite pulse}.dropzone .dz-preview .dz-progress{opacity:1;z-index:1000;pointer-events:none;position:absolute;height:16px;left:50%;top:50%;margin-top:-8px;width:80px;margin-left:-40px;background:rgba(255,255,255,.9);-webkit-transform:scale(1);border-radius:8px;overflow:hidden}.dropzone .dz-preview .dz-progress .dz-upload{background:#333;background:-webkit-gradient(linear,left top,left bottom,from(#666),to(#444));background:linear-gradient(to bottom,#666,#444);position:absolute;top:0;left:0;bottom:0;width:0;-webkit-transition:width .3s ease-in-out;transition:width .3s ease-in-out}.dropzone .dz-preview.dz-error .dz-error-message{display:block}.dropzone .dz-preview.dz-error:hover .dz-error-message{opacity:1;pointer-events:auto}.dropzone .dz-preview .dz-error-message{pointer-events:none;z-index:1000;position:absolute;display:none;opacity:0;-webkit-transition:opacity .3s;transition:opacity .3s;border-radius:8px;font-size:13px;top:130px;left:-10px;width:140px;background:#be2626;background:-webkit-gradient(linear,left top,left bottom,from(#be2626),to(#a92222));background:linear-gradient(to bottom,#be2626,#a92222);padding:.5em 1.2em;color:#fff}.dropzone .dz-preview .dz-error-message:after{content:'';position:absolute;top:-6px;left:64px;width:0;height:0;border-left:6px solid transparent;border-right:6px solid transparent;border-bottom:6px solid #be2626}"]
            }),
            __param(0, core.Inject(core.PLATFORM_ID))
        ], DropzoneComponent);
        return DropzoneComponent;
    }());

    var DropzoneModule = /** @class */ (function () {
        function DropzoneModule() {
        }
        DropzoneModule = __decorate([
            core.NgModule({
                imports: [common.CommonModule],
                declarations: [DropzoneComponent, DropzoneDirective],
                exports: [common.CommonModule, DropzoneComponent, DropzoneDirective]
            })
        ], DropzoneModule);
        return DropzoneModule;
    }());

    exports.DROPZONE_CONFIG = DROPZONE_CONFIG;
    exports.DropzoneComponent = DropzoneComponent;
    exports.DropzoneConfig = DropzoneConfig;
    exports.DropzoneDirective = DropzoneDirective;
    exports.DropzoneModule = DropzoneModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
